JSON/req-schema.json

```diff
@@ -0,0 +1,140 @@
+                }
+            },
+            "additionalProperties": false
+        },
+        "units": {
+            "type": "object",
+            "patternProperties": {
+                "^[A-Za-z]+[A-Za-z0-9_-]*$": {
+                    "$ref": "#/definitions/unit_obj"
+                }
+            },
+            "additionalProperties": false
+        }
+    },
+    "required": [ "validations" ],
+    "definitions": {
+        "mv_param": {
+            "type": "object",
+            "properties": {
+                "arg": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "vals": {
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "minLength": 1
+                    }
+                },
+                "role": {
+                    "type": "string",
+                    "enum": ["client", "server", "all"]
+                },
+                "id": {
+                    "type": "string",
+                    "pattern": "^[1-9][0-9]*$"
+                },
+                "enabled": {
+                    "type": "string",
+                    "enum": ["yes", "no"]
+                }
+            },
+            "required": [
+                "arg",
+                "vals"
+            ],
+            "additionalProperties": false
+        },
+        "validation_obj": {
+            "type": "object",
+            "properties": {
+                "args": {
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "minLength": 1
+                    }
+                },
+                "vals": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "description": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "convert": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "transform": {
+                    "type": "object",
+                    "properties": {
+                        "search": {
+                            "type": "string"
+                        },
+                         "replace": {
+                            "type": "string"
+                        }
+                    },
+                    "required": [ "search", "replace" ]
+                }
+            },
+            "required": [
+                "args",
+                "vals"
+            ],
+            "additionalProperties": false
+        },
+        "unit_obj": {
+            "type": "object",
+            "minItems": 0,
```

Comment:
does it make sense to create a `unit_obj` that is empty?  shouldn't that case be covered by making the `unit_obj` optional above?

Reply:
`units` is an object, not an array, so this is wrong. Changing to:
```
"minProperties": 1
```

Reply:
oops...missed that part myself